### PR TITLE
Elaborate and fix metricq version string

### PR DESCRIPTION
**Pull request by phijor**
_TL;DR: git-builds now have a version string like `"metricq-cpp/r500.g5a7781d"`. Makes answering the question "Which git commit of `metricq` was this build from?` a lot easier._

Use `${CMAKE_CURRENT_SOURCE_DIR` instead of `${CMAKE_SOURCE_DIR}` when deducing version strings from git.  This fixes version strings of parent project including metricq as a submodule having *their* version show up in `metricq::version()`; which is all kinds of wrong.

Additionally, we tweak the version string format to make it easier to figure out from which source commit/release a client was built. 
If the *current* CMake source directory is cointained in a .git repository, we try and generate a version string from git history, which includes the latest metricq commit hash.
If not, we'll simply use `$metricq_VERSION` as defined in `CMakeLists.txt`.

The general format of this is either (in a non-git directory, e.g. a release tarball):

```
  v$MAJOR.$MINOR[.$PATCH]
```

or (when building in a git directory or git submodule)

```
  r$REV_COUNT.g$HEAD_REV
```

where `$REV_COUNT` is the number of revisions since the root commit (`git rev-list --count HEAD`), and `$HEAD_REV` is the (abbreviated) commit hash of latest commit.
Any competent™ version comparison tool should be able to deduce that "`r42.g1234abc` < `v1.0`".

In the future, when there is a `git tag v1.0`, version strings of in-git builds should be formatted as

```
  v$MAJOR.$MINOR[.$PATCH].r$REV_SINCE_LAST_VER.g$HEAD_REV[-dirty]
```

similar to what `git describe --long --tags --dirty` outputs.

### CODE REVIEW

**Comment by bmario** [comment]
As we discussed, this isn't needed anymore.
